app/models/distribution.rb

```diff
@@ -40,10 +41,10 @@ class Distribution < ApplicationRecord
   validate :line_items_exist_in_inventory
   validate :line_items_quantity_is_positive
 
-  before_save :combine_distribution
+  before_save :combine_distribution, :reset_shipping_cost, :validate_shipping_cost
```

Comment:
You should keep validations as validations rather than callbacks.

```ruby
validates :shipping_cost, :numericality => { :greater_than_or_equal_to => 0 }
```